src/main/scala/com/cognite/sdk/scala/v1/resources/functions.scala

```diff
@@ -0,0 +1,192 @@
+import com.softwaremill.sttp._
+import com.softwaremill.sttp.circe._
+import io.circe.{Decoder, Encoder}
+import io.circe.derivation.{deriveDecoder, deriveEncoder}
+import io.circe.Json
+import io.circe.Printer
+
+class Functions[F[_]](val requestSession: RequestSession[F])
+    extends WithRequestSession[F]
+    with Create[Function, FunctionCreate, F]
+    with RetrieveByIds[Function, F]
+    with RetrieveByExternalIds[Function, F]
+    with DeleteByIds[F, Long]
+    with DeleteByExternalIds[F] {
+  import Functions._
+  override val baseUrl = uri"${requestSession.baseUrl}/functions"
+
+  def read(): F[Items[Function]] =
+    requestSession.get[Items[Function], Items[Function]](
+      baseUrl,
+      value => value
+    )
+
+  override def createItems(items: Items[FunctionCreate]): F[Seq[Function]] = {
+    implicit val customPrinter: Printer = Printer.noSpaces.copy(dropNullValues = true)
+    requestSession.post[Seq[Function], Items[Function], Items[FunctionCreate]](
+      items,
+      baseUrl,
+      value => value.items
+    )
+  }
+
+  override def deleteByIds(ids: Seq[Long]): F[Unit] =
+    DeleteByIds.deleteByIds(requestSession, baseUrl, ids)
+
+  override def deleteByExternalIds(externalIds: Seq[String]): F[Unit] =
+    DeleteByExternalIds.deleteByExternalIds(requestSession, baseUrl, externalIds)
+
+  override def retrieveByIds(ids: Seq[Long]): F[Seq[Function]] =
+    RetrieveByIds.retrieveByIds(requestSession, baseUrl, ids)
+
+  override def retrieveByExternalIds(externalIds: Seq[String]): F[Seq[Function]] =
+    RetrieveByExternalIds.retrieveByExternalIds(requestSession, baseUrl, externalIds)
+}
+
+object Functions {
+  implicit val functionErrorDecoder: Decoder[FunctionError] = deriveDecoder[FunctionError]
+  implicit val functionDecoder: Decoder[Function] = deriveDecoder[Function]
+  implicit val functionsItemsDecoder: Decoder[Items[Function]] =
+    deriveDecoder[Items[Function]]
+  implicit val functionsItemsWithCursorDecoder: Decoder[ItemsWithCursor[Function]] =
+    deriveDecoder[ItemsWithCursor[Function]]
+  implicit val functionErrorEncoder: Encoder[FunctionError] = deriveEncoder[FunctionError]
+  implicit val createFunctionEncoder: Encoder[FunctionCreate] = deriveEncoder[FunctionCreate]
+  implicit val createFunctionsItemsEncoder: Encoder[Items[FunctionCreate]] =
+    deriveEncoder[Items[FunctionCreate]]
+
+  implicit val errorOrFunctionDecoder: Decoder[Either[CdpApiError, Items[Function]]] =
+    EitherDecoder.eitherDecoder[CdpApiError, Items[Function]]
+  implicit val errorOrUnitDecoder: Decoder[Either[CdpApiError, Unit]] =
+    EitherDecoder.eitherDecoder[CdpApiError, Unit]
+  implicit val deleteRequestWithRecursiveAndIgnoreUnknownIdsEncoder
+      : Encoder[ItemsWithRecursiveAndIgnoreUnknownIds] =
+    deriveEncoder[ItemsWithRecursiveAndIgnoreUnknownIds]
+}
+
+class FunctionCalls[F[_]](val requestSession: RequestSession[F], val functionId: Long)
+    extends WithRequestSession[F]
+    with BaseUrl {
```

Comment:
Couldn't this also use some standard traits like Readable, RetrieveByIds, Filter?

Reply:
Function calls only has retrieveById and does not support multiple ids. Also, none of the function endpoints support cursor and limit.